http/src/request/channel/invite/create_invite.rs

```diff
@@ -62,19 +72,29 @@ impl<'a> CreateInvite<'a> {
     ///
     /// If no age is specified, Discord sets the age to 86400 seconds, or 24 hours.
     /// Set to 0 to never expire.
-    pub fn max_age(mut self, max_age: u64) -> Self {
+    pub fn max_age(mut self, max_age: u64) -> Result<Self, CreateInviteError> {
+        if max_age > 604_800 {
+            return Err(CreateInviteError::MaxAgeInvalid(
+                "max_age should be between 0 and 604800.".to_string(),
+            ));
+        }
         self.fields.max_age.replace(max_age);
 
-        self
+        Ok(self)
     }
 
     /// Set the maximum uses for an invite, or 0 for infinite.
     ///
     /// Discord defaults this to 0, or infinite.
-    pub fn max_uses(mut self, max_uses: u64) -> Self {
+    pub fn max_uses(mut self, max_uses: u64) -> Result<Self, CreateInviteError> {
```

Comment:
Changing the return signature of this function is a breaking change, so this needs to target the `v0.4` branch. It might be necessary to close this PR and open another based off of `v0.4`.